rapid7_insightvm/komand_rapid7_insightvm/util/resource_requests.py

```diff
@@ -0,0 +1,203 @@
+            payload = dict()
+        if not params:
+            params = {
+                'size': 500,
+                'page': current_page
+            }
+        if isinstance(params, list):
+            parameters = RequestParams.from_tuples(params)
+        else:
+            parameters = RequestParams.from_dict(params)
+
+        while True:
+            self.logger.info(f'Fetching results from page {current_page}')
+            parameters['page'] = current_page
+            if number_of_results != 0:
+                if results_retrieved + parameters['size'] > number_of_results:
+                    parameters['size'] = number_of_results - results_retrieved
+                    last_page = True
+
+            response = self.get_resource_page(endpoint=endpoint,
+                                              method=method,
+                                              params=parameters,
+                                              payload=payload)
+
+            resources += response.resources  # Grab resources and append to total
+            self.logger.info(f'Got {len(response.resources)} resources '
+                             f'from page {(response.page_num+1)} / {response.total_pages}')
+
+            if (response.total_pages == 0) or ((response.total_pages - 1) == response.page_num):
+                self.logger.info("All pages consumed, returning results...")
+                break  # exit the loop
+            elif last_page:
+                self.logger.info(f'{number_of_results} results consumed, returning results')
+                break
+            else:
+                self.logger.info('More pages exist, fetching...')
+                current_page += 1
+
+        return resources
+
+    def get_resource_page(self, endpoint: str, method: str, params: RequestParams, payload: dict) -> RequestResult:
+        """
+        Retrieves resources from a security console
+        :param endpoint: Endpoint to reach
+        :param method: HTTP method for the API request
+        :param params: URL parameters to append to the request
+        :param payload: JSON body for the API request if required
+        :return: Namedtuple object containing current page number, total pages, and results
+        """
+        # Get size and page from list of dict param type
+
+        self.logger.info(f'Fetching up to {params["size"]} resources from endpoint page {params["page"]} ...')
+        request_method = getattr(self.session, method.lower())
+
+        extras = {"json": payload, "params": params.params}
+        try:
+            response = request_method(url=endpoint, verify=False, **extras)
+        except requests.RequestException as e:
+            assistance = self._REQUEST_EXCEPTIONS.get(type(e), self._UNHANDLED_EXCEPTION)
+            raise PluginException(cause=e, assistance=assistance)
+
+        resource_request_status_code_check(response.text, response.status_code)
+        response_json = response.json()
```

Comment:
Try/except around this for possible JSONDecodeError - and PluginException preset